File: Gestion_de_Musica_BACKEND/Models/AlbumModel.cs
using Gestion_de_Musica.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;


namespace Gestion_de_Musica.Models
{
    public class Album
    {
        [Key]
        public int album_id { get; set; }

        [Required]
        [StringLength(200)]
        public string titulo { get; set; } = string.Empty;

        [StringLength(100)]
        public string? genero { get; set; }

        public int? año_lanzamiento { get; set; }

        [StringLength(200)]
        public string? discografica { get; set; }

        public int? artista_id { get; set; }

        public DateTime fecha_creacion { get; set; } = DateTime.Now;

        // Navegación
        public Artista? Artista { get; set; }
    }
}

File: Gestion_de_Musica/src/app/models/album.model.ts
export interface Album {
  album_id: number;
  titulo: string;
  genero?: string;
  año_lanzamiento?: number;
  discografica?: string;
  artista_id?: number;
  nombre_artista?: string;
}

export interface CreateAlbum {
  titulo: string;
  genero?: string;
  año_lanzamiento?: number;
  discografica?: string;
  artista_id?: number;
}
